protocol/bc/outputcommitment.go

```diff
@@ -0,0 +1,66 @@
+package bc
+
+import (
+	"fmt"
+	"io"
+
+	"chain/encoding/blockchain"
+	"chain/encoding/bufpool"
+	"chain/errors"
+)
+
+// OutputCommitment contains the commitment data for a transaction
+// output (which also appears in the spend input of that output).
+type OutputCommitment struct {
+	AssetAmount
+	VMVersion      uint64
+	ControlProgram []byte
+}
+
+func (oc *OutputCommitment) writeTo(w io.Writer, assetVersion uint64) (err error) {
+	b := bufpool.Get()
```

Comment:
I ended up leaving OutputCommitment alone in this “code moving” PR - I find sometimes it’s hard to figure out what’s going on when you move code into a new file _and_ make changes, and I definitely tied myself up a bit when I was trying to move this code and rewrite it at the same time. But let me know if it seems gratuitous to have this PR on its own without further changes.